bftengine/src/bftengine/messages/ViewChangeMsg.cpp

```diff
@@ -162,7 +162,9 @@ void ViewChangeMsg::finalizeMessage() {
   // |               Message Body               |
   // +------------------------------------------+
 
-  sigManager->sign(body(), bodySize, body() + bodySize);
+  // TODO: since there are multiple message with different sequences, make sure that lastStable()
```

Comment:
AFAIU if a replica is signing with its latest known sequence number of the view, then it is the correct cryptosystem for sure.
Assuming the last stable is X for replica R. so the R's working window is in between [X, X + 300]. Also, we do know that R is synced up to X. 
The only gap is if the other replicas have committed R's new key between [X, X + 300], but even so, the key won't be effective before X + 600.
If X is too far from the current working window, then R will first perform state transfer, sync with the state (and the correct key), and then the above also holds.
Hence, I think the code is correct and the TODO can be removed.

Reply:
removed